File: submissions/0007-reverse-integer/solution.java
public class Solution {
    public int reverse(int x) {
        boolean isMinus = 0L > x;
        if (isMinus) {
            x = Math.abs(x);
            if(0L > x) {
                return 0;
            }
        }
        char[] newChars = reverseCharArray(String.valueOf(x).toCharArray());
        Long l = Long.parseLong((isMinus ? "-" : "") + String.valueOf(newChars));
        if(l > Integer.MAX_VALUE || l < Integer.MIN_VALUE) {
            return 0;
        } else {
            return l.intValue();
        }
    }

    private char[] reverseCharArray(char[] chars) {
        char[] newChars = new char[chars.length];
        for (int i = 0; i < chars.length; i++) {
            newChars[i] = chars[chars.length - 1 - i];
        }
        return newChars;
    }
}

File: submissions/0007-reverse-integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        result = 0

        isNegative = x < 0
        x = abs(x)

        while x > 0:
            digit = x % 10
            x //= 10
            result *= 10
            result += digit

        if isNegative:
            result *= -1

        if result > 2**31 -1 or result < -2**31:
            return 0
        
        return result
